Modified references to reflection pickle files from dials to use msgpack

jmp1985 opened the pull request:
This pull request is related to the changes made in dials to use msgpack rather than pickle to serialize reflection data. https://github.com/dials/dials/pull/743

I have made some changes here to make sure that reflection files are serialized in the same way. Old reflection files in the pickle format can still be read using dials programs and if you used the dials.util.option.OptionParser class then pickle and msgpack files will be read transparently anyway.

ndevenish commented on xfel/command_line/cspad_cbf_metrology.py:
```diff
@@ -67,7 +67,7 @@
     .help = Algorithm to be used for choosing the n_subset images/experiments for \
             refinement.  n_refl chooses the set with the largest numbers of reflections \
-            listed in the pickle files, thus giving maximal coverage of the detector tiles \
+            listed in the mpack files, thus giving maximal coverage of the detector tiles \
```
suggest using "reflection table" for things like this, then it's independent of whatever format

phyy-nx commented:
Hi, I'm out for a week, so I won't be able to review until next Monday.  One comment though.  I haven't looked at any of the changes, but I'm hoping that you've created an abstract 'deserialization' class that can read pickles and mpack files.

jmp1985 commented:
Hi @phyy-nx. All the dials programs using the dials.util.options.OptionParser class accept both pickle and msgpack files. In terms of reading in files manually I have added a function that accepts both to the reflection table.

However, within cctbx.xfel there was only one bit I needed to change that imports a pickle file, most changes were for default file names and for exporting files. I have changed the line to use a function that can read in both pickle and msgpack.

Also @ndevenish I've modified the comment

phyy-nx commented on xfel/command_line/cspad_cbf_metrology.py:
```diff
@@ -244,9 +244,9 @@ def find_files(path, reflections):
   for filename in os.listdir(path):
     if reflections in filename:
-      exp_path = os.path.join(path, filename.rstrip("_%s.pickle"%reflections) + "_refined_experiments.json")
+      exp_path = os.path.join(path, filename.rstrip("_%s.mpack"%reflections) + "_refined_experiments.json")
```
Needs to look for pickles or mpack files for backwards compatibility

phyy-nx commented on xfel/command_line/cspad_cbf_metrology.py:
```diff
@@ -244,9 +244,9 @@ def find_files(path, reflections):
+      exp_path = os.path.join(path, filename.rstrip("_%s.mpack"%reflections) + "_refined_experiments.json")
       if not os.path.exists(exp_path):
-        exp_path = os.path.join(path, filename.rstrip("_%s.pickle"%reflections) + "_experiments.json")
+        exp_path = os.path.join(path, filename.rstrip("_%s.mpack"%reflections) + "_experiments.json")
```
Needs to look for pickles or mpack files for backwards compatibility

phyy-nx commented on xfel/command_line/striping.py:
```diff
@@ -264,7 +264,7 @@ def allocate_chunks(results_dir,
                     max_size=1000,
                     integrated=False):
-  refl_ending = "_integrated.pickle" if integrated else "_indexed.pickle"
+  refl_ending = "_integrated.mpack" if integrated else "_indexed.mpack"
```
Need to look for pickle files here too.
